destination/destination.go

```diff
@@ -0,0 +1,57 @@
+package destination
```

Comment:
this package will be shared across xcodebuil test-related steps, I think we should find a better place to share this logic, I was thinking about go-steputils or go-xcode.

Reply:
Yes, I was about to move it to `go-xcode`, but in a separate PR, as the Xcode Test migration is already taking long.